ignite/contrib/handlers/clearml_logger.py

```diff
@@ -806,7 +806,11 @@ def get_local_copy(self, filename: str) -> Optional[str]:
 
     @idist.one_rank_only()
     def remove(self, filename: str) -> None:
-        super(ClearMLSaver, self).remove(filename)
+        from clearml.storage.helper import StorageHelper
+
+        clearml_logger = self._task.get_logger()
+        helper = StorageHelper.get(filename, logger=clearml_logger)
+        helper.delete(filename)
```

Comment:
@devrimcavusoglu could you please explain why you are adding this code ?
I understand that the idea is to use clearml's v1.0 new feature and make `remove` work as expected for others.
I'm not sure how it would work with the previous logic where a queue is used...

cc @bmartinn any suggestion on how we could recode everything here ?

Reply:
As you said, this would make use of clearml v1.0 new feature, and used to actually remove the remote or local file(s). For backward compatibility, I'd appreciate any suggestions.

Reply:
No need to pass `logger` to `StorageHelper.get` , basically this should be fine:
```python
helper = StorageHelper.get(filename)
helper.delete(filename)
```

Reply:
Thanks for the review & comment @bmartinn. I'll update PR with that change asap. I and @vfdev-5 were mainly concerned about backward compatibility as it may negatively impact users using `clearml<1.0`; however, I see that it is not a concern.

Reply:
@devrimcavusoglu I rechecked backwards compatibility,  the only exception will be uploading models directly to `clearml-server` < 1.0. If this is the case the delete function might return an error as the http file server did not support the delete operation. I really think that just protecting with `except ValueError` should be enough. wdyt?
```python
try:
  helper.delete(filename)
except ValueError:
  # log something ?!
```

Reply:
@bmartinn Thanks for reviewing. Well, yes it should be sufficient I believe. As we protect the older implementation `super().remove()` inside `StorageHelper`. 

With the suggested change, all things should be okay. I'll add a warning (for `clearml-server<1.0` users) to point out for this.

Reply:
@bmartinn can you please confirm whether we still need `_CallbacksContext` and `_checkpoint_slots` code with clearml v1.X ?

I have an impression that if we could remove data from remote storage then we can drastically simplify `ClearMLSaver` for v1.X ?

Reply:
@vfdev-5 Hi there, sorry for delayed reply (was in tight schedule). Option 2 seems more solid to me, and also more practical. Regarding the class seperation (former comment) for different versions, I think rather than pointing to V0 and V1, handling it under `ClearMLSaver` would be more concrete, wdyt ?

Reply:
> Regarding the class seperation (former comment) for different versions, I think rather than pointing to V0 and V1, handling it under ClearMLSaver would be more concrete, wdyt ?

I was thinking that doing things into two separate classes will help us later to quickly remove V0 when no one is using it anymore...

Reply:
Hi @vfdev-5,

> @bmartinn can you please confirm whether we still need _CallbacksContext and _checkpoint_slots code with clearml v1.X ?

I think that if you handle storing a new checkpoint to a temporary-named newly uploaded, file, removing the previously uploaded file and renaming the temporary-named file to the previous name, you won't need the checkpoints 🙂